grakn-graql/src/main/java/ai/grakn/graql/internal/query/InsertQueryExecutor.java

```diff
@@ -268,8 +247,32 @@ private Concept putConceptByType(Optional<String> name, Optional<String> id, Var
                 Pattern rhs = var.getProperty(RhsProperty.class).get().getRhs();
                 return type.asRuleType().addRule(lhs, rhs);
             });
+        } else if (type.getName().equals(Schema.MetaSchema.CONCEPT.getName())) {
+            throw new IllegalStateException(var + " cannot be an instance of meta-type " + type.getName());
```

Comment:
This might be a redundant check because core already ensures that meta's are immutable.